include/buoy_msgs/interface.hpp

```diff
@@ -482,17 +537,23 @@ class Interface : public rclcpp::Node
             pbsrv_enum2str[future.get()->result.value].c_str());
           // TODO(andermi): should we shutdown?
         }
+        // TODO(anyone) Why do I need this assignment after every `async_send_request` callback?
+        // Without this, results in `std::bad_function_call` on the second `async_send_request`
+        this->pc_wind_curr_callback = service_response_callback<PCWindCurrServiceCallback,
+            PCWindCurrServiceResponseFuture>();
```

Comment:
@chapulina @quarkytale I can't figure out why I need to do this. It's really puzzling me. I don't think a `std::function` after being initialized:
`std::function<...(...)> = [...](...) -> ... {...};`
should become initialized after being called successfully once. If I initialize it after every use, I don't get `std::bad_function_call`

Reply:
example usage of `async_send_request` and `pc_wind_curr_callback`

https://github.com/osrf/buoy_examples/blob/060ee1b078f2e790df3a74d97f8c405af81e9486/src/torque_controller.cpp#L35

Reply:
@andermi , do you have step-by-step instructions on how we should run these to test them?

Reply:
@chapulina Well, for the moment, it runs fine with lines 542-543. Perhaps we could open an issue/PR to see about removing this and merge this PR in for now?

Reply:
I've been looking at this for a while and still have no clue about what's happening. The only thing I can think of is refactoring the code not to store these callbacks as member variables.

We only need to set `pc_wind_curr_callback` here because that's the one that happens to be used in the example, right? So if someone tries to use a different one, they will run into the same issue.

I'm ok merging this to unblock things, but I think we should get back to it soon.

Reply:
At the moment, all of the callbacks are basically no-ops. They all do the same thing, and print the result of the service call. Basically just a default since I didn't care about the result. The error message is helpful for debugging code, but not really important in production. What might be important is if we lose comms and care to do something about it. There doesn't really seem to be much to do except quit or wait for the connection to come back. We could even just use the version of the service call without a callback in the example case.
If something needs to happen synchronously with various service calls, just waiting on the `future` is also valid.

Reply:
I just possibly found something... see #15

Reply:
@chapulina maybe see if you like the latest version that removes the #15 issue

Reply:
LGTM 👍🏽